types/src/common.ts

```diff
@@ -1,20 +1,55 @@
-import { Struct, Option, Text, bool, Vec, u16, u32, u64, Null } from '@polkadot/types'
-import { BlockNumber, Moment } from '@polkadot/types/interfaces'
-import { Codec, RegistryTypes } from '@polkadot/types/types'
+import { Struct, Option, Text, bool, Vec, u16, u32, u64, Null, U8aFixed, BTreeSet, Compact } from '@polkadot/types'
+import { BlockNumber, H256, Moment } from '@polkadot/types/interfaces'
+import { Codec, Constructor, InterfaceTypes, Registry, RegistryTypes } from '@polkadot/types/types'
+import { u8aConcat, u8aToHex } from '@polkadot/util'
 // we get 'moment' because it is a dependency of @polkadot/util, via @polkadot/keyring
 import moment from 'moment'
 import { JoyStructCustom, JoyStructDecorated } from './JoyStruct'
 import { JoyEnum } from './JoyEnum'
 
 export { JoyEnum, JoyStructCustom, JoyStructDecorated }
 
+// Adds ".sort()" during BTreeSet toU8a encoding (required by the runtime)
+// FIXME: Will not cover cases where BTreeSet is part of extrinsic args metadata
+export class JoyBTreeSet<V extends Codec> extends BTreeSet<V> {
+  public static with<V extends Codec>(valType: Constructor<V> | keyof InterfaceTypes): Constructor<JoyBTreeSet<V>> {
+    return class extends JoyBTreeSet<V> {
+      constructor(registry: Registry, value?: Uint8Array | string | Set<any>) {
+        super(registry, valType, value)
+      }
+    }
+  }
+
+  public toU8a(isBare?: boolean): Uint8Array {
+    const encoded = new Array<Uint8Array>()
+
+    if (!isBare) {
+      encoded.push(Compact.encodeU8a(this.size))
+    }
+
+    Array.from(this)
+      .sort()
+      .forEach((v: V) => {
+        encoded.push(v.toU8a(isBare))
+      })
+
+    return u8aConcat(...encoded)
+  }
+
+  public toHex(): string {
+    return u8aToHex(this.toU8a())
+  }
+}
+
 export class Credential extends u64 {}
 export class CredentialSet extends Vec.with(Credential) {} // BtreeSet ?
 
 // common types between Forum and Proposal Discussions modules
 export class ThreadId extends u64 {}
 export class PostId extends u64 {}
 
+export class Hash extends U8aFixed implements H256 {}
```

Comment:
This is just a "helper" for cases when class constructor is needed instead of an interface (`@polkadot/types` only expose `Hash` interface, while the underlying class in `U8aFixed`) 
`Hash` and `H256` interfaces should be identical, but I can implement the original one for more clarity.

Reply:
Ok thanks for clarification. I think H256 more explicitly matches the length 32-bit length we use in our runtime so its fine.